scripts/analyze_csv_data.py

```diff
@@ -481,19 +483,22 @@ def performance_ratio_fn(df, par):
         df.groupby(["instance_name"])
         .apply(lambda _: performance_ratio_fn(_, par))
         .reset_index(name="ratio")
+        .dropna()
     )
     plt.figure(figsize=(10, 6))
     plt.title(
         sanitize_title(
             f"{prefix} {xlabel} {label_lookup(l0)}:{label_lookup(l1)}"
         )
     )
-    plot_loghist(ratios["ratio"], min(len(ratios) // 3, 25))
-    path = os.path.join(
-        FIGS_DIR,
-        f"{prefix}_{label_lookup(l0)}_{label_lookup(l1)}_{xaxis}_performance_ratio.pdf",
-    )
-    plt.savefig(path)
+    nbins = min(len(ratios) // 3, 25)
+    if nbins > 0:
```

Comment:
Move these two lines up to before plt.figure (line 488), right after computing ratios, just to save creating and discarding the plt.figure().